### ISSUE TITLE
Shuffle repeating playlist songs randomly

### ISSUE THREAD

**Issue by flkrz** [lucaspape/catplayer#29]
**Describe the bug**
When playing a playlist on shuffle, sometimes tracks will get played twice, despite not being at the end of the playlist. 

**To Reproduce**
1. Go to playlists
2. Click on context menu and play next on a playlist
3. Open main player window
4. Ensure shuffle and playlist repeat enabled
5. Play tracks normally or skip through, behaviour is the same

**Expected behavior**
Each track plays once in turn through the whole playlist before starting again (if repeating playlist is enabled)

**Screenshots**
http://severas.rm-ni.uk:8080/C.A.S./req0018-20200228/Screen_Recording_20200228-121146.mp4

**Smartphone (please complete the following information):**
 - Device: Samsung Galaxy Note 9
 - OS: Android 10 with OneUI2.0
 - Browser N/A
 - Version 0.1.0.2

**Additional context**
Similar behaviour noticed with previous builds, but low priority at the time.

**Comment by lucaspape** [comment]
Yes i know about that, to be honest I just were too lazy to implement it correctly. It is on my list

**Comment by lucaspape** [comment]
I now have a workaround for this which is absolutely awful (but it works really well)

**Comment by flkrz** [comment]
Do I want to know how you hacked this to work out just trust that it won't cause my phone to explode? ;)

**Comment by lucaspape** [comment]
Well, I just created a list in which every song is listed that was already played in random mode. If the next song is in this list it will just try a random song again.

**Comment by lucaspape** [comment]
This should now be fixed properly, I now rewrote the entire player class (should work way better, I am testing it a little bit now, I will also probably add crossfade now)
